sbt-plugin/src/main/scala/org/scalajs/sbtplugin/AbstractJSDeps.scala

```diff
@@ -75,6 +75,11 @@ object ProvidedJSModuleID {
     ProvidedJSModuleID(new JSDependency(name, Nil), configurations)
 }
 
+/** Deprecated.
+ *
+ *  For internal reasons, this class is not marked as `@deprecated`, but should
```

Comment:
Same here: What are "internal reasons". Please document (internally). Here it seems OK that the users don't get a warning, since this is probably not really used externally.